File: src/Listas3/pratica/Questao02.java
package Listas3.pratica;

import java.util.Scanner;

class ValidarSenha {
    String senha;

    public ValidarSenha(String senha) {
        this.senha = senha;
    }

    public void verificarMaiusculas() throws IllegalArgumentException {
        for (int i = 0; i < this.senha.length(); i++) {
            if (Character.isUpperCase(this.senha.charAt(i))) {
                return;
            }
        }
        throw new IllegalArgumentException("Erro: a senha deve conter pelo menos uma letra maiúscula");
    }
}

public class Questao02 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        while (true) {
            String entrada = sc.nextLine();
            try {
                ValidarSenha senha1 = new ValidarSenha(entrada);
                senha1.verificarMaiusculas();
                System.out.println("Senha válida");
                break;
            } catch (IllegalArgumentException e) {
                System.out.println(e.getMessage());
            }
        }
        sc.close();
    }
}

File: Listas C++/Lista 3/questao02.cpp
#include <iostream>
#include <string>
#include <cctype>
#include <stdexcept>

class ValidarSenha {
    std::string senha;

public:
    ValidarSenha(std::string senha) {
        this->senha = senha;
    }

    void verificarMaiusculas() {
        for (char c : this->senha) {
            if (std::isupper(c)) {
                return;
            }
        }
        throw std::invalid_argument("Erro: a senha deve conter pelo menos uma letra maiúscula");
    }
};

int main() {
    std::string entrada;
    
    while (true) {
        if (!std::getline(std::cin, entrada)) {
            break;
        }

        try {
            ValidarSenha senha1(entrada);
            senha1.verificarMaiusculas();
            std::cout << "Senha válida" << std::endl;
            break;
        } catch (const std::invalid_argument& e) {
            std::cout << e.what() << std::endl;
        }
    }

    return 0;
}